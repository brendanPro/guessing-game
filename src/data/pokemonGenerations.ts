export interface PokemonGeneration {
  id: number;
  name: string;
  startId: number;
  endId: number;
  region: string;
  description: string;
}

export const POKEMON_GENERATIONS: PokemonGeneration[] = [
  {
    id: 1,
    name: "Génération I",
    startId: 1,
    endId: 151,
    region: "Kanto",
    description: "Les 151 premiers Pokémon (Rouge, Bleu, Jaune)"
  },
  {
    id: 2,
    name: "Génération II", 
    startId: 1,
    endId: 251,
    region: "Johto",
    description: "Les 251 premiers Pokémon (Or, Argent, Cristal)"
  },
  {
    id: 3,
    name: "Génération III",
    startId: 1,
    endId: 386,
    region: "Hoenn",
    description: "Les 386 premiers Pokémon (Rubis, Saphir, Émeraude)"
  },
  {
    id: 4,
    name: "Génération IV",
    startId: 1,
    endId: 493,
    region: "Sinnoh",
    description: "Les 493 premiers Pokémon (Diamant, Perle, Platine)"
  },
  {
    id: 5,
    name: "Génération V",
    startId: 1,
    endId: 649,
    region: "Unys",
    description: "Les 649 premiers Pokémon (Noir, Blanc, Noir 2, Blanc 2)"
  },
  {
    id: 6,
    name: "Génération VI",
    startId: 1,
    endId: 721,
    region: "Kalos",
    description: "Les 721 premiers Pokémon (X, Y, Rubis Oméga, Saphir Alpha)"
  },
  {
    id: 7,
    name: "Génération VII",
    startId: 1,
    endId: 809,
    region: "Alola",
    description: "Les 809 premiers Pokémon (Soleil, Lune, Ultra-Soleil, Ultra-Lune)"
  },
  {
    id: 8,
    name: "Génération VIII",
    startId: 1,
    endId: 905,
    region: "Galar",
    description: "Les 905 premiers Pokémon (Épée, Bouclier, Légendes Arceus)"
  },
  {
    id: 9,
    name: "Génération IX",
    startId: 1,
    endId: 1025,
    region: "Paldea",
    description: "Tous les Pokémon (Écarlate, Violet, DLC)"
  }
];

export const DEFAULT_GENERATION = POKEMON_GENERATIONS[0]; // Gen I by default
