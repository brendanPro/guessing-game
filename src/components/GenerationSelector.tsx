import React from 'react';
import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
import { Button } from '@/components/ui/button';
import { cn } from '@/lib/utils';
import { type PokemonGeneration, POKEMON_GENERATIONS } from '@/data/pokemonGenerations';

interface GenerationSelectorProps {
  selectedGeneration: PokemonGeneration;
  onGenerationChange: (generation: PokemonGeneration) => void;
  disabled?: boolean;
  className?: string;
}

export function GenerationSelector({ 
  selectedGeneration, 
  onGenerationChange, 
  disabled = false,
  className 
}: GenerationSelectorProps) {

  return (
    <Card className={cn("w-full max-w-4xl mx-auto", className)}>
      <CardHeader>
        <CardTitle className="text-center text-lg">
          üéÆ Choisissez une G√©n√©ration
        </CardTitle>
        <p className="text-center text-sm text-muted-foreground">
          S√©lectionnez la g√©n√©ration de Pok√©mon pour votre partie
        </p>
      </CardHeader>
      <CardContent>
        <div className="grid grid-cols-2 sm:grid-cols-3 md:grid-cols-4 lg:grid-cols-5 xl:grid-cols-9 gap-3">
          {POKEMON_GENERATIONS.map((gen) => (
            <Button
              key={gen.id}
              variant={selectedGeneration.id === gen.id ? "default" : "outline"}
              size="sm"
              onClick={() => onGenerationChange(gen)}
              disabled={disabled}
              className={cn(
                "flex flex-col items-center p-3 h-auto min-h-[80px] sm:min-h-[60px]",
                "hover:scale-105 transition-transform touch-manipulation",
                selectedGeneration.id === gen.id && "ring-2 ring-primary"
              )}
            >
              <div className="font-bold text-xs sm:text-xs">{gen.name}</div>
              <div className="text-xs text-muted-foreground">{gen.startId}-{gen.endId}</div>
              <div className="text-xs text-muted-foreground hidden sm:block">{gen.region}</div>
            </Button>
          ))}
        </div>
        
        <div className="mt-4 p-3 bg-muted rounded-lg">
          <div className="text-sm font-semibold">
            {selectedGeneration.name} - {selectedGeneration.region}
          </div>
          <div className="text-xs text-muted-foreground">
            {selectedGeneration.description}
          </div>
        </div>
      </CardContent>
    </Card>
  );
}
