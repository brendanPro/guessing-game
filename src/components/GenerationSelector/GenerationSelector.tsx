import React from 'react';
import { Label } from '@/components/ui/label';
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from '@/components/ui/select';
import { cn } from '@/lib/utils';
import { type PokemonGeneration, POKEMON_GENERATIONS } from '@/data/pokemonGenerations';

interface GenerationSelectorProps {
  selectedGeneration: PokemonGeneration;
  onGenerationChange: (generation: PokemonGeneration) => void;
  disabled?: boolean;
  className?: string;
}

export function GenerationSelector({ 
  selectedGeneration, 
  onGenerationChange, 
  disabled = false,
  className 
}: GenerationSelectorProps) {
  const handleValueChange = (value: string) => {
    const generation = POKEMON_GENERATIONS.find(gen => gen.id.toString() === value);
    if (generation) {
      onGenerationChange(generation);
    }
  };

  return (
    <div className={cn("space-y-3", className)}>
      <div className="space-y-2">
        <Label htmlFor="generation-select">Génération Pokémon</Label>
        <Select
          value={selectedGeneration.id.toString()}
          onValueChange={handleValueChange}
          disabled={disabled}
        >
          <SelectTrigger id="generation-select" className="w-full">
            <SelectValue placeholder="Choisissez une génération" />
          </SelectTrigger>
          <SelectContent>
            {POKEMON_GENERATIONS.map((gen) => (
              <SelectItem key={gen.id} value={gen.id.toString()}>
                <div className="flex items-center justify-between w-full gap-2">
                  <span className="font-medium">{gen.name}</span>
                  <span className="text-xs text-muted-foreground">
                    {gen.region} (#{gen.startId}-{gen.endId})
                  </span>
                </div>
              </SelectItem>
            ))}
          </SelectContent>
        </Select>
      </div>
      
      <div className="p-3 bg-muted rounded-lg text-sm">
        <div className="font-semibold">
          {selectedGeneration.name} - {selectedGeneration.region}
        </div>
        <div className="text-xs text-muted-foreground mt-1">
          {selectedGeneration.description}
        </div>
      </div>
    </div>
  );
}
