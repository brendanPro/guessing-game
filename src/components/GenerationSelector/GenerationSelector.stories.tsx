import "@/index.css"
import type { Meta, StoryObj } from '@storybook/react';
import { GenerationSelector } from './GenerationSelector';
import { POKEMON_GENERATIONS } from '@/data/pokemonGenerations';
import { fn } from '@storybook/test';
import { useState } from 'react';

const meta = {
  title: 'Game/GenerationSelector',
  component: GenerationSelector,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    selectedGeneration: {
      control: 'object',
      description: 'The currently selected generation',
    },
    onGenerationChange: {
      action: 'generation changed',
      description: 'Callback when a generation is selected',
    },
    disabled: {
      control: 'boolean',
      description: 'Whether the selector is disabled',
    },
  },
  args: {
    onGenerationChange: fn(),
  },
} satisfies Meta<typeof GenerationSelector>;

export default meta;
type Story = StoryObj<typeof meta>;

export const GenerationI: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[0],
    disabled: false,
  },
};

export const GenerationII: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[1],
    disabled: false,
  },
};

export const GenerationV: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[4],
    disabled: false,
  },
};

export const GenerationIX: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[8],
    disabled: false,
  },
};

export const Disabled: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[0],
    disabled: true,
  },
};

export const Interactive: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[0],
    disabled: false,
  },
  render: (args) => {
    const [selected, setSelected] = useState(args.selectedGeneration);
    
    return (
      <div className="w-full max-w-4xl">
        <GenerationSelector
          {...args}
          selectedGeneration={selected}
          onGenerationChange={(gen) => {
            setSelected(gen);
            args.onGenerationChange?.(gen);
          }}
        />
        <div className="mt-4 p-4 bg-muted rounded-lg text-center">
          <p className="text-sm">
            <strong>Selected:</strong> {selected.name} ({selected.region})
          </p>
          <p className="text-xs text-muted-foreground">
            Pok√©mon #{selected.startId} - #{selected.endId}
          </p>
        </div>
      </div>
    );
  },
};

export const AllGenerations: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[0],
    disabled: false,
  },
  render: (args) => (
    <div className="space-y-4">
      <h3 className="text-lg font-semibold text-center">All 9 Generations</h3>
      <GenerationSelector {...args} />
    </div>
  ),
};

export const WithCustomWidth: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[0],
    disabled: false,
  },
  render: (args) => (
    <div className="w-[600px]">
      <GenerationSelector {...args} />
    </div>
  ),
};

export const MobileView: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[0],
    disabled: false,
  },
  parameters: {
    viewport: {
      defaultViewport: 'mobile1',
    },
  },
};

export const TabletView: Story = {
  args: {
    selectedGeneration: POKEMON_GENERATIONS[0],
    disabled: false,
  },
  parameters: {
    viewport: {
      defaultViewport: 'tablet',
    },
  },
};

