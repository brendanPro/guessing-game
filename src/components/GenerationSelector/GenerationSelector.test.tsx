import { describe, test, expect, mock } from "bun:test";
import { renderToStaticMarkup } from "react-dom/server";
import { GenerationSelector } from "./GenerationSelector";
import { POKEMON_GENERATIONS } from '@/data/pokemonGenerations';


describe("GenerationSelector Component", () => {
  const mockOnChange = mock(() => {});

  test("snapshot with Generation I selected", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Generation V selected", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[4]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toMatchSnapshot();
  });

  test("snapshot with Generation IX selected", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[8]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toMatchSnapshot();
  });

  test("snapshot when disabled", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={true}
      />
    );
    expect(html).toMatchSnapshot();
  });

  test("contains label", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toContain("Génération Pokémon");
  });

  test("has select element", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toContain("generation-select");
  });


  test("displays selected generation description", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toContain("Les 151 premiers Pokémon (Rouge, Bleu, Jaune)");
  });

  test("displays selected generation name in description box", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toContain("Génération I - Kanto");
  });

  test("has description box", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toContain("bg-muted");
  });

  test("has space-y layout", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toContain("space-y-3");
  });

  test("applies custom className when provided", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[0]} 
        onGenerationChange={mockOnChange}
        disabled={false}
        className="custom-class"
      />
    );
    expect(html).toContain("custom-class");
  });

  test("displays Generation V correctly", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[4]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toContain("Génération V - Unys");
    expect(html).toContain("Les 649 premiers Pokémon");
  });

  test("displays Generation IX correctly", () => {
    const html = renderToStaticMarkup(
      <GenerationSelector 
        selectedGeneration={POKEMON_GENERATIONS[8]} 
        onGenerationChange={mockOnChange}
        disabled={false}
      />
    );
    expect(html).toContain("Génération IX - Paldea");
    expect(html).toContain("Tous les Pokémon");
  });

});


