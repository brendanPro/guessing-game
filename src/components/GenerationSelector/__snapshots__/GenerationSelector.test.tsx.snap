// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerationSelector Component snapshot with Generation I selected 1`] = `"<div class="space-y-3"><div class="space-y-2"><label data-slot="label" class="text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50" for="generation-select">Génération Pokémon</label><button type="button" role="combobox" aria-controls="radix-_R_9_" aria-expanded="false" aria-autocomplete="none" dir="ltr" data-state="closed" data-slot="select-trigger" class="border-input data-[placeholder]:text-muted-foreground aria-invalid:border-destructive ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 [&amp;_svg:not([class*=&#x27;text-&#x27;])]:text-muted-foreground flex h-9 items-center justify-between rounded-md border bg-transparent px-3 py-2 text-sm shadow-xs transition-[color,box-shadow] focus-visible:ring-4 focus-visible:outline-1 disabled:cursor-not-allowed disabled:opacity-50 aria-invalid:focus-visible:ring-0 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&amp;_svg]:pointer-events-none [&amp;_svg]:shrink-0 [&amp;_svg:not([class*=&#x27;size-&#x27;])]:size-4 [&amp;&gt;span]:line-clamp-1 w-full" id="generation-select"><span data-slot="select-value" style="pointer-events:none"></span><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-chevron-down size-4 opacity-50" aria-hidden="true"><path d="m6 9 6 6 6-6"></path></svg></button><select aria-hidden="true" tabindex="-1" style="position:absolute;border:0;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0, 0, 0, 0);white-space:nowrap;word-wrap:normal"></select></div><div class="p-3 bg-muted rounded-lg text-sm"><div class="font-semibold">Génération I - Kanto</div><div class="text-xs text-muted-foreground mt-1">Les 151 premiers Pokémon (Rouge, Bleu, Jaune)</div></div></div>"`;

exports[`GenerationSelector Component snapshot with Generation V selected 1`] = `"<div class="space-y-3"><div class="space-y-2"><label data-slot="label" class="text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50" for="generation-select">Génération Pokémon</label><button type="button" role="combobox" aria-controls="radix-_R_9_" aria-expanded="false" aria-autocomplete="none" dir="ltr" data-state="closed" data-slot="select-trigger" class="border-input data-[placeholder]:text-muted-foreground aria-invalid:border-destructive ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 [&amp;_svg:not([class*=&#x27;text-&#x27;])]:text-muted-foreground flex h-9 items-center justify-between rounded-md border bg-transparent px-3 py-2 text-sm shadow-xs transition-[color,box-shadow] focus-visible:ring-4 focus-visible:outline-1 disabled:cursor-not-allowed disabled:opacity-50 aria-invalid:focus-visible:ring-0 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&amp;_svg]:pointer-events-none [&amp;_svg]:shrink-0 [&amp;_svg:not([class*=&#x27;size-&#x27;])]:size-4 [&amp;&gt;span]:line-clamp-1 w-full" id="generation-select"><span data-slot="select-value" style="pointer-events:none"></span><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-chevron-down size-4 opacity-50" aria-hidden="true"><path d="m6 9 6 6 6-6"></path></svg></button><select aria-hidden="true" tabindex="-1" style="position:absolute;border:0;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0, 0, 0, 0);white-space:nowrap;word-wrap:normal"></select></div><div class="p-3 bg-muted rounded-lg text-sm"><div class="font-semibold">Génération V - Unys</div><div class="text-xs text-muted-foreground mt-1">Les 649 premiers Pokémon (Noir, Blanc, Noir 2, Blanc 2)</div></div></div>"`;

exports[`GenerationSelector Component snapshot with Generation IX selected 1`] = `"<div class="space-y-3"><div class="space-y-2"><label data-slot="label" class="text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50" for="generation-select">Génération Pokémon</label><button type="button" role="combobox" aria-controls="radix-_R_9_" aria-expanded="false" aria-autocomplete="none" dir="ltr" data-state="closed" data-slot="select-trigger" class="border-input data-[placeholder]:text-muted-foreground aria-invalid:border-destructive ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 [&amp;_svg:not([class*=&#x27;text-&#x27;])]:text-muted-foreground flex h-9 items-center justify-between rounded-md border bg-transparent px-3 py-2 text-sm shadow-xs transition-[color,box-shadow] focus-visible:ring-4 focus-visible:outline-1 disabled:cursor-not-allowed disabled:opacity-50 aria-invalid:focus-visible:ring-0 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&amp;_svg]:pointer-events-none [&amp;_svg]:shrink-0 [&amp;_svg:not([class*=&#x27;size-&#x27;])]:size-4 [&amp;&gt;span]:line-clamp-1 w-full" id="generation-select"><span data-slot="select-value" style="pointer-events:none"></span><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-chevron-down size-4 opacity-50" aria-hidden="true"><path d="m6 9 6 6 6-6"></path></svg></button><select aria-hidden="true" tabindex="-1" style="position:absolute;border:0;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0, 0, 0, 0);white-space:nowrap;word-wrap:normal"></select></div><div class="p-3 bg-muted rounded-lg text-sm"><div class="font-semibold">Génération IX - Paldea</div><div class="text-xs text-muted-foreground mt-1">Tous les Pokémon (Écarlate, Violet, DLC)</div></div></div>"`;

exports[`GenerationSelector Component snapshot when disabled 1`] = `"<div class="space-y-3"><div class="space-y-2"><label data-slot="label" class="text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50" for="generation-select">Génération Pokémon</label><button type="button" role="combobox" aria-controls="radix-_R_9_" aria-expanded="false" aria-autocomplete="none" dir="ltr" data-state="closed" disabled="" data-disabled="" data-slot="select-trigger" class="border-input data-[placeholder]:text-muted-foreground aria-invalid:border-destructive ring-ring/10 dark:ring-ring/20 dark:outline-ring/40 outline-ring/50 [&amp;_svg:not([class*=&#x27;text-&#x27;])]:text-muted-foreground flex h-9 items-center justify-between rounded-md border bg-transparent px-3 py-2 text-sm shadow-xs transition-[color,box-shadow] focus-visible:ring-4 focus-visible:outline-1 disabled:cursor-not-allowed disabled:opacity-50 aria-invalid:focus-visible:ring-0 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&amp;_svg]:pointer-events-none [&amp;_svg]:shrink-0 [&amp;_svg:not([class*=&#x27;size-&#x27;])]:size-4 [&amp;&gt;span]:line-clamp-1 w-full" id="generation-select"><span data-slot="select-value" style="pointer-events:none"></span><svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-chevron-down size-4 opacity-50" aria-hidden="true"><path d="m6 9 6 6 6-6"></path></svg></button><select aria-hidden="true" tabindex="-1" disabled="" style="position:absolute;border:0;width:1px;height:1px;padding:0;margin:-1px;overflow:hidden;clip:rect(0, 0, 0, 0);white-space:nowrap;word-wrap:normal"></select></div><div class="p-3 bg-muted rounded-lg text-sm"><div class="font-semibold">Génération I - Kanto</div><div class="text-xs text-muted-foreground mt-1">Les 151 premiers Pokémon (Rouge, Bleu, Jaune)</div></div></div>"`;
